### PR TITLE
add synchronization and traces for cleanup operation

### PR DESCRIPTION
**Pull request by suwatch**

### CODE REVIEW

File: Kudu.Core/Tracing/XmlTracer.cs [hunk 1]
```diff
@@ -238,30 +246,61 @@ public static bool ShouldTrace(TraceLevel level, TraceInfo info, int statusCode)
...
+                    {
+                        var files = FileSystemHelpers.GetFiles(_path, "*.xml");
+                        if (files.Length < MaxXmlFiles)
+                        {
+                            return;
+                        }
+
+                        var toCleanup = files.Length - (MaxXmlFiles * 80) / 100;
+                        var attribs = new Dictionary<string, string>
+                    {
```
Comment (davidebbo) [inline Kudu.Core/Tracing/XmlTracer.cs]: Indent

**Comment by davidebbo** [comment]
:shipit: other than minor cleanup discussed

File: Kudu.Core/Tracing/XmlTracer.cs [hunk 2]
```diff
@@ -98,7 +101,7 @@ private IDisposable WriteStartElement(string title, IDictionary<string, string>
 
                 strb.Append(new String(' ', _infos.Count * 2));
                 strb.AppendFormat("<step title=\"{0}\" ", XmlUtility.EscapeXmlText(title));
-                strb.AppendFormat("date=\"{0}\" ", DateTime.UtcNow.ToString("yyy-MM-ddTHH:mm:ss.fff"));
+                strb.AppendFormat("date=\"{0}\" ", info.StartTime.ToString("yyy-MM-ddTHH:mm:ss.fff"));
```
Comment (ahmelsayed) [inline Kudu.Core/Tracing/XmlTracer.cs]: I know this is the way it was before the PR, but shouldn't it be 4 Ys as in `"yyyy-MM-ddTHH:mm:ss.fff"` vs `"yyy-MM-ddTHH:mm:ss.fff"` like in [here](https://github.com/projectkudu/kudu/pull/1714/files?diff=unified#diff-27786ae8673c1540926cc9debfd30b31L277)

Comment (suwatch) [inline Kudu.Core/Tracing/XmlTracer.cs]: yes it should.   good catch.  will fix it.

**Comment by davidebbo** [comment]
4b7dc6a3158af155c0157e88101a43f629291e10

**Pull request closed without merging**